### PR TITLE
refactor: introduce CSigBase which becomes the base class for CRecoveredSig, CSigShare and CSigSesAnn

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
…redSig, CSigShare and CSigSesAnn

### CODE REVIEW

File: src/llmq/signing_shares.h [hunk 1]
```diff
@@ -393,7 +412,7 @@ class CSigSharesManager : public CRecoveredSigsListener
     void ProcessMessage(const CNode* pnode, const std::string& strCommand, CDataStream& vRecv);
 
     void AsyncSign(const CQuorumCPtr& quorum, const uint256& id, const uint256& msgHash);
-    CSigShare CreateSigShare(const CQuorumCPtr& quorum, const uint256& id, const uint256& msgHash) const;
+    std::optional<CSigShare> CreateSigShare(const CQuorumCPtr& quorum, const uint256& id, const uint256& msgHash) const;
```
Comment (UdjinM6) [inline src/llmq/signing_shares.h:415]: probably needs `#include <optional>` like in #4781

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/llmq/signing_shares.h (hunk 1 above)
Comment (PastaPastaPasta) [reply src/llmq/signing_shares.h:415]: :D

Comment (thephez) [reply src/llmq/signing_shares.h:416]: Thanks @UdjinM6. @PastaPastaPasta is always trying to break my builds :smirk:

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**